File: strategy/DuckSimulator/python/behaviors.py
from abc import ABC, abstractmethod


class flyBehavior(ABC):
	@abstractmethod
	def fly(self):
		pass


class FlyLow(flyBehavior):
	def fly(self):
		print("Flying low")


class FlyHigh(flyBehavior):
	def fly(self):
		print("Flying high")

class FlyMoon(flyBehavior):
	def fly(self):
		print("Flying to the moon")


class quackBehavior(ABC):
	@abstractmethod
	def quack(self):
		pass


class quackNoisy(quackBehavior):
	def quack(self):
		print("QUACK QUACK")

class quackQuietly(quackBehavior):
	def quack(self):
		print("quack quack")

File: strategy/DuckSimulator/cpp/behaviors.cpp
#include <iostream>
#include "behaviors.h"

void FlyLow::fly() const {
	std::cout << "Flying low" << std::endl;
}

void FlyHigh::fly() const {
	std::cout << "Flying High" << std::endl;
}

void FlyFast::fly() const {
	std::cout << "Flying Fast" << std::endl;
}

void QuackNoisy::quack() const {
	std::cout << "QUACK QUACK" << std::endl;
}

void QuackQuietly::quack() const {
	std::cout << "quack quack" << std::endl;
}